### PR TITLE
[9.x] Artisan `about` Command

### PR DESCRIPTION
**Pull request by jbrooksuk**
This PR introduces a new `about` command that displays output about the system environment and configuration of the user's application. The command is inspired by Rails' own [`about`](https://guides.rubyonrails.org/v2.3/command_line.html#about) command, though it's beautiful and has some additional powers 🪄 

CleanShot 2022-07-13 at 12 08 46@2x

The command includes its own API so that applications and packages may push data into existing sections or create a new section with its own data:

```php
use Illuminate\Foundation\Console\AboutCommand;

AboutCommand::add('Laravel Vapor', 'Vapor Runtime', '1.0.0');
```

You can also add multiple items:

```php
AboutCommand::add('Environment', [
    'Laravel Cashier' => 'Stripe',
    'Laravel Nova' => '4.0.0'
]);
```

It's expected that packages will start to add their own sections of information, which could make this command output a lot of information. You can choose to filter down certain sections with `--only` and pass a comma-separated list of sections in `snake_case`.

CleanShot 2022-07-13 at 12 10 13@2x

Finally, you can choose to output this information as JSON by using the `--json` option (which can also be combined with `--only`):

CleanShot 2022-07-13 at 12 11 40@2x

### CODE REVIEW

File: src/Illuminate/Foundation/Console/AboutCommand.php [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+     * @deprecated
+     */
+    protected static $defaultName = 'about';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'About your application\'s environment';
```
Comment (nunomaduro) [inline src/Illuminate/Foundation/Console/AboutCommand.php:35]: It seems a little bit more than the Application's environment at this point?

File: src/Illuminate/Support/Composer.php [hunk 2]
```diff
@@ -107,4 +107,21 @@ public function setWorkingPath($path)
...
     }
+
+    /**
+     * Get the version of Composer.
+     *
+     * @return string|null
+     */
+    public function getVersion()
+    {
+        $command = array_merge($this->findComposer(), ['-V']);
```
Comment (nunomaduro) [inline src/Illuminate/Support/Composer.php:118]: @jbrooksuk Did you manage to test this with different composer versions?

**Comment by danharrin** [comment]
This is great, for package maintainers, it will allow us to more easily get debugging information from our users, which will help us fix issues. Not just version numbers, but checking if they have fallen into common traps like not linking their public storage directory.

Very nice!

File: src/Illuminate/Foundation/Console/AboutCommand.php [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    /**
+     * Gather data about the application's environment.
+     *
+     * @return array
+     */
+    protected function gatherEnvironmentData()
+    {
+        static::add('Environment', [
+            'Laravel Version' => app()->version(),
```
Comment (sebdesign) [inline src/Illuminate/Foundation/Console/AboutCommand.php:107]: Why not `$this->laravel->version()`?

Comment (jbrooksuk) [reply src/Illuminate/Foundation/Console/AboutCommand.php]: I've switched to using this `$this->laravel` for consistency.

**Comment by pxlrbt** [comment]
For sites with many packages, I think it might be useful to filter results. For example `php artisan about --only=environment,laravel_nova`

**Comment by voidgraphics** [comment]
Great PR, it's a little hard to read though with the values being so far from the keys. Especially in full screen.

**Comment by fgaroby** [comment]
@pxlrbt: wouldn't it be better using the `vendor/package_name` syntax ?

**Comment by jbrooksuk** [comment]
Okay, I've added `--only` which accepts a command-separated list of section names in snake case, i.e. `Laravel Nova => laravel_nova`.

I've also added an `about` method to `ServiceProvider` as a shortcut to adding sections.

**Comment by jbrooksuk** [comment]
> Great PR, it's a little hard to read though with the values being so far from the keys. Especially in full screen.

The about is dynamic based on the width of your terminal:

CleanShot 2022-07-13 at 10 56 40@2x

**Comment by jbrooksuk** [comment]
We'll address this in the Console UI Improvements PR, on which this PR is based on.

**Comment by dennisprudlo** [comment]
@jbrooksuk Just a thought: How about getting rid of the `--only` option and use an optional argument list instead?
So `about {sections?*} {--json}`. This way the command reads better:
```shell
php artisan about environment
php artisan about environment laravel_nova some_other_section --json
```

**Comment by omerbaflah** [comment]
Great work!  @jbrooksuk  👏🏻

**Pull request merged**